packages/material-ui/src/utils/integerPropType.js

```diff
@@ -0,0 +1,63 @@
+function getTypeByValue(value) {
+  const valueType = typeof value;
+  switch (valueType) {
+    case 'number':
+      if (Number.isNaN(value)) {
```

Comment:
Interesting. This is not supported in IE 11. However, I see that we started to use it in

https://github.com/mui-org/material-ui/blob/a628fc2ccb6a9ec85ce87de83acdd8d6bb3339e3/packages/material-ui-unstyled/src/Unstable_TrapFocus/Unstable_TrapFocus.js#L27

While we were doing this before:

https://github.com/mui-org/material-ui/blob/a628fc2ccb6a9ec85ce87de83acdd8d6bb3339e3/packages/material-ui/src/Tabs/Tabs.js#L211-L213

Reply:
I think that we can keep Number.isNaN as it would only trigger if you use IE 11 as your primary dev environment. We can ignore this case.